native/src/main/java/io/xlibb/docreader/DocReader.java

```diff
@@ -0,0 +1,139 @@
+ */
+public class DocReader {
+
+    private static final Tika tika = new Tika();
+
+    /**
+     * Parses a document file and extracts its type, extension, and content.
+     */
+    public static Object parseDocument(BString filePath, BTypedesc typed) {
+        try {
+            String filePathStr = filePath.getValue();
+            File file = new File(filePathStr);
+            
+            if (!file.exists()) {
+                return createError("File does not exist: " + filePathStr);
+            }
+            
+            if (!file.isFile()) {
+                return createError("Path is not a file: " + filePathStr);
+            }
+
+            // Extract file extension
+            String extension = extractFileExtension(file);
+
+            // Detect MIME type and extract content
+            String mimeType = tika.detect(file);
+            String content = extractContent(file);
+
+            // Create and return Ballerina DocumentInfo record
+            return createDocumentInfo(mimeType, extension, content, typed);
+            
+        } catch (IOException | TikaException | SAXException e) {
+            return createError("Error parsing document: " + e.getMessage());
+        } catch (Exception e) {
+            return createError("Unexpected error: " + e.getMessage());
+        }
+    }
+
+    private static String extractFileExtension(File file) {
+        Path path = Paths.get(file.getAbsolutePath());
+        Path fileNamePath = path.getFileName();
+        
+        if (fileNamePath != null) {
+            String fileName = fileNamePath.toString();
+            int lastDotIndex = fileName.lastIndexOf('.');
+            if (lastDotIndex > 0 && lastDotIndex < fileName.length() - 1) {
+                return fileName.substring(lastDotIndex + 1).toLowerCase(Locale.ENGLISH);
+            }
+        }
+        return "";
+    }
+
+    private static String extractContent(File file) throws IOException, TikaException, SAXException {
+        BodyContentHandler handler = new BodyContentHandler(-1);
+        AutoDetectParser parser = new AutoDetectParser();
+        Metadata metadata = new Metadata();
+        ParseContext context = new ParseContext();
+
+        PDFParserConfig pdfConfig = new PDFParserConfig();
+        pdfConfig.setExtractInlineImages(false);
+        pdfConfig.setExtractUniqueInlineImagesOnly(true);
+        context.set(PDFParserConfig.class, pdfConfig);
+
+        try (FileInputStream inputStream = new FileInputStream(file)) {
+            parser.parse(inputStream, handler, metadata, context);
+            return handler.toString().trim();
+        }
+    }
+
+    private static BMap<BString, Object> createDocumentInfo(String docType, String extension, 
+            String content, BTypedesc typed) {
+        DocumentInfo docInfo = new DocumentInfo(
+            docType != null ? docType : "application/octet-stream", 
+            extension, 
+            content
+        );
```

Comment:
This logic seems redundant. We can directly use the function’s arguments to create the Ballerina record, no need to maintain a java record.